### PR TITLE
python312Packages.qtpy: normalize pname

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #245383

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 295143` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.openllm</li>
    <li>python311Packages.openllm.dist</li>
    <li>python311Packages.stytra</li>
    <li>python311Packages.stytra.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>16 packages failed to build:</summary>
  <ul>
    <li>electrum-ltc</li>
    <li>electrum-ltc.dist</li>
    <li>napari (python311Packages.napari)</li>
    <li>napari.dist (python311Packages.napari.dist)</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.glueviz</li>
    <li>python311Packages.glueviz.dist</li>
    <li>python311Packages.pvextractor</li>
    <li>python311Packages.pvextractor.dist</li>
    <li>python312Packages.magicgui</li>
    <li>python312Packages.magicgui.dist</li>
    <li>python312Packages.napari-npe2</li>
    <li>python312Packages.napari-npe2.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>113 packages built:</summary>
  <ul>
    <li>corrscope</li>
    <li>corrscope.dist</li>
    <li>electron-cash</li>
    <li>electron-cash.dist</li>
    <li>electrum</li>
    <li>electrum-grs</li>
    <li>electrum-grs.dist</li>
    <li>electrum.dist</li>
    <li>flent</li>
    <li>flent.dist</li>
    <li>git-cola</li>
    <li>git-cola.dist</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkcut.dist</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>linien-gui</li>
    <li>linien-gui.dist</li>
    <li>natron</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-kernel.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.echo</li>
    <li>python311Packages.echo.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.enaml</li>
    <li>python311Packages.enaml.dist</li>
    <li>python311Packages.enamlx</li>
    <li>python311Packages.enamlx.dist</li>
    <li>python311Packages.encodec</li>
    <li>python311Packages.encodec.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.jupyter</li>
    <li>python311Packages.jupyter.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.magicgui</li>
    <li>python311Packages.magicgui.dist</li>
    <li>python311Packages.napari-console</li>
    <li>python311Packages.napari-console.dist</li>
    <li>python311Packages.napari-npe2</li>
    <li>python311Packages.napari-npe2.dist</li>
    <li>python311Packages.nbdev</li>
    <li>python311Packages.nbdev.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python311Packages.phik</li>
    <li>python311Packages.phik.dist</li>
    <li>python311Packages.pyannote-audio</li>
    <li>python311Packages.pyannote-audio.dist</li>
    <li>python311Packages.pydeck</li>
    <li>python311Packages.pydeck.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>python311Packages.qdarkstyle</li>
    <li>python311Packages.qdarkstyle.dist</li>
    <li>python311Packages.qreactor</li>
    <li>python311Packages.qreactor.dist</li>
    <li>python311Packages.qtawesome</li>
    <li>python311Packages.qtawesome.dist</li>
    <li>python311Packages.qtconsole</li>
    <li>python311Packages.qtconsole.dist</li>
    <li>python311Packages.qtpy</li>
    <li>python311Packages.qtpy.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.rotary-embedding-torch</li>
    <li>python311Packages.rotary-embedding-torch.dist</li>
    <li>spyder (python311Packages.spyder)</li>
    <li>spyder.dist (python311Packages.spyder.dist)</li>
    <li>streamlit (python311Packages.streamlit)</li>
    <li>streamlit.dist (python311Packages.streamlit.dist)</li>
    <li>python311Packages.superqt</li>
    <li>python311Packages.superqt.dist</li>
    <li>python311Packages.vega</li>
    <li>python311Packages.vega.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python311Packages.xarray-einstats</li>
    <li>python311Packages.xarray-einstats.dist</li>
    <li>python312Packages.enaml</li>
    <li>python312Packages.enaml.dist</li>
    <li>python312Packages.nbdev</li>
    <li>python312Packages.nbdev.dist</li>
    <li>python312Packages.qdarkstyle</li>
    <li>python312Packages.qdarkstyle.dist</li>
    <li>python312Packages.qtpy</li>
    <li>python312Packages.qtpy.dist</li>
    <li>quarto</li>
    <li>rare</li>
    <li>rare.dist</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>smb3-foundry</li>
  </ul>
</details>
